test/e2e/deployment.go

```diff
@@ -433,6 +446,8 @@ func testRolloverDeployment(f *framework.Framework) {
 
 	// Wait for replica set to become ready before adopting it.
 	framework.Logf("Waiting for pods owned by replica set %q to become ready", rsName)
+	By("Wait for pods become ready")
+	Expect(framework.WaitForPodsReady(c, ns, podName, 10)).NotTo(HaveOccurred())
```

Comment:
WaitForReadyReplicaSet should be enough to use. If all pods are ready then the replica set should have its status updated eventually.